presto-hive/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftMetastoreUtil.java

```diff
@@ -435,7 +435,7 @@ private static OptionalLong fromMetastoreDistinctValuesCount(OptionalLong distin
     private static long fromMetastoreDistinctValuesCount(long distinctValuesCount, long nullsCount, long rowCount)
     {
         long nonNullsCount = rowCount - nullsCount;
-        if (nullsCount > 0) {
+        if (nullsCount > 0 && (distinctValuesCount > 1 || nonNullsCount == 0)) {
```

Comment:
Actually it should be just `nullsCount > 0 && distinctValuesCount > 0`. If `nonNullsCount  == 0` - the number of distinct values should be zero.

Reply:
I think it actually might be:
```
if (nullsCount > 0 && (distinctValuesCount > 1))
```
We don't want to subtract if `NDV` is 1.